code/test-utils/src/test/java/com/adobe/marketing/tester/util/JSONObjectAssertsTests.kt

```diff
@@ -0,0 +1,80 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+package com.adobe.marketing.tester.util
+
+import com.adobe.marketing.tester.util.JSONObjectAsserts.assertTypeMatch
+import org.json.JSONObject
+import org.junit.Assert.assertThrows
+import org.junit.Test
+
+class JSONObjectAssertsTests {
```

Comment:
can you include all the tests here similar to the ios implementation?
I also noticed an issue with the alternate mode paths, the assert passes when it shouldn't